tls/s2n_alerts.c

```diff
@@ -145,8 +145,15 @@ static bool s2n_alerts_supported(struct s2n_connection *conn)
 
 static bool s2n_process_as_warning(struct s2n_connection *conn, uint8_t level, uint8_t type)
 {
-    /* Only TLS1.2 considers the alert level. The alert level field is
-     * considered deprecated in TLS1.3. */
+    /*
+     *= https://tools.ietf.org/rfc/rfc8446#section-6
+     *# All the alerts listed in Section 6.2 MUST be sent with
+     *# AlertLevel=fatal and MUST be treated as error alerts when received
+     *# regardless of the AlertLevel in the message.
+     *
+     * Only TLS1.2 considers the alert level. The alert level field is
+     * considered deprecated in TLS1.3.
+     */
     if (s2n_connection_get_protocol_version(conn) < S2N_TLS13) {
```

Comment:
This still doesn't seem right to me. You've put a TLS1.3 compliance comment on a chunk of code that is specifically "<  S2N_TLS13". 

Maybe move it to be a comment on the method instead? Or maybe include the TLS1.2 requirement that the level be respected?

Reply:
It's also a bit of an issue that this only enforces one of the two MUSTs in this statement. From the compliance report, it'll look like we're enforcing both :/

Reply:
I couldnt find a concrete tls1.2 requirement so went with the tls13 one. I can move it to the top of the method.

I would like to keep the context of `all alerts` so wondering if this looks ok. Its unfortunate we have to split up the citations and lose context.

```
+/*
+ * In TLS1.3 all alerts
+ *= https://tools.ietf.org/rfc/rfc8446#section-6
+ *# MUST be treated as error alerts when received
+ *# regardless of the AlertLevel in the message.
+ */
```